torch/utils/data/datapipes/utils/common.py

```diff
@@ -20,6 +21,60 @@
 ]
 
 
+def validate_input_col(fn: Callable, input_col: Optional[Union[int, tuple, list]]):
+    """
+    Checks that function used in a map style datapipe works with the input column
+
+    Args:
+        fn: The function to check.
+        input_col: The input column to check.
+    Returns:
+        None.
+    Raises:
+        TypeError: If the function is not compatible with the input column.
+    """
+    sig = inspect.signature(fn)
+    if isinstance(input_col, (list, tuple)):
+        input_col_size = len(input_col)
+    else:
+        input_col_size = 1
+
+    fn_name = str(fn)
+
+    def get_non_default_params(signature: inspect.Signature):
+        excluded_types = [inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD]
+        return [p for p in signature.parameters.values() if p.default is p.empty and p.kind not in excluded_types]
+
+    if len(sig.parameters) > input_col_size:
+        non_default_params = get_non_default_params(sig)
+
+        if len(non_default_params) > input_col_size:
+
+            raise ValueError(
+                f"The function {fn_name} takes {len(non_default_params)} "
+                f"non-default parameters, but {input_col_size} are required."
+            )
+
+        non_default_kw_only = [p for p in non_default_params if p.kind == inspect.Parameter.KEYWORD_ONLY]
+
+        if len(non_default_kw_only) > 0:
+            raise ValueError(
+                f"The function {fn_name} takes {len(non_default_kw_only)} "
+                f"non-default keyword-only parameters, which is not allowed."
+            )
+
```

Comment:
I forgot to handle the case where a function could have 0 positional args but 2 kw-only args, which will still fail.